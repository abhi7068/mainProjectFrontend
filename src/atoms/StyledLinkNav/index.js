import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
const StyledLinks = styled(Link)`
  text-decoration: none;
  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
    color: black;
  }
`;
const StyledLink = (props) => {
  return (
    <div>
      <StyledLinks data-testid="StyledLink " to={props.to}>
        {' '}
        {props.innerText}
      </StyledLinks>
    </div>
  );
};
StyledLink.propTypes = {
  to: PropTypes.string,
  innerText: PropTypes.string,
};

export default StyledLink;
