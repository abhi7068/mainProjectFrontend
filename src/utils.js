import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
