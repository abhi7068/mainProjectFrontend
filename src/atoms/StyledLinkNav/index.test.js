import React from 'react';
import { render, cleanup } from '../../utils';
import StyledLinkNav from '../StyledLinkNav/index';

afterEach(cleanup);
it('checkinginnerText', () => {
  const { getByTestId } = render(
    <StyledLinkNav innerText="Home" to="/"></StyledLinkNav>
  );
  expect(getByTestId('StyledLink')).toHaveTextContent('Home');
});
it('creatingsnapshot', () => {
  const { asFragment } = render(
    <StyledLinkNav innerText="Login" to="/Login"></StyledLinkNav>
  );
  expect(asFragment()).toMatchSnapshot();
});
