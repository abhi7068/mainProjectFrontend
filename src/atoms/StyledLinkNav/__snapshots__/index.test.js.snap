// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creatingsnapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="sc-AxjAm eutjKG"
      data-testid="StyledLink "
      href="/Login"
    >
       Login
    </a>
  </div>
</DocumentFragment>
`;
